src/eliater/regression.py

```diff
@@ -0,0 +1,502 @@
+    data.mean()['Y']
+
+This means that, intervention on $X$ and fixing its value to 0 causes a decrease in the expected value of $Y$. Now
+we will explore the output of the query in the form of ATE:
+$\mathbb{E}[Y \mid do(X=x+1)] - \mathbb{E}[Y \mid do(X=x+1)]$. Note that the value of $x$ is indifferent in the
+output, because ATE amounts to the estimate for the coefficient of $X$ (\hat{\lambda_{xy}}), in a regression equation
+where $X$, and the optimal adjustment set $Z_3$ are regressed on $Y$ as follows:
+
+$Y = \lambda_{xy} X + \lambda_{z_3Y} Z_3 + U_Y; U_Y \sim \mathcal{N}(0, \sigma^2_Y)$
+
+Note that, instead of $Z_3$ in the equation above, one could have used $Z_1$ or $Z_2$, or any combination of $Z_i$ for
+$i \in \{1,2,3\}$. However, this module is designed such that if the optimal adjustment set exists, it picks it. Hence,
+in this example $Z_3$ is selected and used.
+
+.. code-block:: python
+
+    from y0.dsl import Variable, X, Y
+    from eliater.frontdoor_backdoor import example_2
+    from eliater.regression import estimate_query
+
+    graph = example_2.graph
+    data = example_2.generate_data(100, seed=100)
+
+    estimate_query(
+        graph=graph,
+        data=data,
+        treatments={X},
+        outcome=Y,
+        query_type="ate"
+    )
+
+The estimated value for ATE (average contrast of Y under two distinct interventions
+on $X$) amounts to 0.123. Finally, we estimate the query in the form of "probability".
+
+.. code-block:: python
+
+    from y0.dsl import Variable, Z, X, Y
+    from eliater.frontdoor_backdoor import example_2
+    from eliater.regression import estimate_query
+
+    graph = example_2.graph
+    data = example_2.generate_data(100, seed=100)
+
+    estimate_query(
+        graph=graph,
+        data=data,
+        treatments={X},
+        outcome=Y,
+        query_type="probability",
+        interventions={X: 0},
+    )
+
+The output is a vector of $Y$ values (100 data points). The mean over these values is equal to the
+expected value obtained above when the query was in the form of "expected value". This type of query
+can be used to get summary statistics for the distribution of $Y$, when $X$ has been intervened upon.
+For example, we can get a summary statistics over $Y$ as follows:
+
+.. code-block:: python
+
+    from y0.dsl import Variable, Z, X, Y
+    from eliater.frontdoor_backdoor import example_2
+    from eliater.regression import estimate_query
+
+    graph = example_2.graph
+    data = example_2.generate_data(100, seed=100)
+    summary_statistics(
+        graph=graph,
+        data=data,
+        treatments={X},
+        outcome=Y,
+        interventions={X: 0},
+    )
+
+The output is as follows:
+
+'''SummaryStatistics(size=100.0, mean=69.65468808666209, std=7.134828083971124, min=51.47221859593694,
+first_quartile=64.48297054774694, second_quartile=69.74351236828335, third_quartile=74.41515094449318,
+max=91.8646554122394)'''
+
+Unanswered Questions for Later
```

Comment:
@srtaheri here are the remaining questions - I want to merge this so we can update the other code to be rerunnable, and these questions will come up then